### ISSUE TITLE
Make icons accessible again

### ISSUE THREAD

**Issue by youdz** [vmware/clarity#2608]
## Describe the bug

Icons have no way of being read by a screenreader, even with a proper title.

## How to reproduce

https://stackblitz.com/edit/icons-accessibility

1.  Turn on your screenreader.
2.  Try to read the home icon in the main content, and fail.

## Expected behavior

If I provide a title for an icon, I expect it to be read by a screenreader.

## Versions

**App**

* Angular: 6
* Clarity: 0.12

## Additional notes

The `aria-hidden="true"` is the most obvious culprit here, but it's also worth noting the `aria-labelledby` is probably on the wrong element, it needs to be on the `<svg>`.
No matter what we go with, we have to remember some icons are presentational, while some actually convey information. For instance:
- icons in a tree of items are presentational, the name of the item is enough information,
- the icon for an alert is the only way to identify the alert's level (warning, success, etc.) outside of the color.

The easiest way would probably be to always make icons readable by screenreaders, and recommend using `role="presentation"` on the app's side for icons that should be skipped over.

**Comment by Shijir** [comment]
@youdz I am not really sure or clear on whether the default macOS screenreader is supposed to select and register the icon's title if it's not wrapped inside an interactive element. Without an interactive wrapping element because it's only decorative icon, not an interactive icon. The default macOS screenreader skips over images with alt text as well unless they are wrapped inside an interactive element (StackBlitz is appended below). So if a Clarity icon is wrapped inside an interacitve element, the title attribute works just fine as the VoiceOver could select and read its title. 

https://stackblitz.com/edit/icons-and-image-accessibility

Also, I think it's worth looking at Material Icons. https://material.angular.io/components/icon/overview#accessibility They classify icons in the categories as such:
>**In thinking about accessibility, it is useful to place icon use into one of three categories:** 
>- **Decorative**: the icon conveys no real semantic meaning and is purely cosmetic.
>- **Interactive**: a user will click or otherwise interact with the icon to perform some action.
>- **Indicator**: the icon is not interactive, but it conveys some information, such as a status. This includes using the icon in place of text inside of a larger message.

Similarly, in our case, without that wrapper interactive element, icons with title attriute will only trigger the native browsers' tooltip when users hover over them and show sigthed users what those icons stand for.

PS: _it could be also that I am using the VoiceOver in a different or wrong way. So I think we should discuss this issue further._

**Comment by coryrylan** [comment]
After more investigation @Shijir is correct with the use cases of the icon.The Material docs look exactly like what the current recommendations are for icon a11y. The only issue I see in the current implementation is the `title` attribute on the `clr-icon` is not setting the `aria-labelledby` on the svg element and removing `aria-hidden` when it is set.
